src/ResendUtil.js

```diff
@@ -0,0 +1,35 @@
+import EventEmitter from 'eventemitter3'
+import { ControlLayer } from 'streamr-client-protocol'
+
+export default class ResendUtil extends EventEmitter {
+    constructor() {
+        super()
+        this.subForRequestId = {}
+        this.counter = 0
+    }
+
+    generateRequestId() {
+        const id = this.counter
+        this.counter += 1
+        return id.toString()
+    }
+
+    getSubFromResendResponse(response, responseTypeName) {
+        if (!this.subForRequestId[response.subId]) { // TODO: replace with response.requestId
+            const error = new Error(`Received unexpected ${responseTypeName} message ${response.serialize()}`)
```

Comment:
Could we use `response.constructor.name` to achieve the same and get rid of one extra parameter?